optimistic-transaction: remove reference to being "default" (#3146)
pingcap/docs: optimistic-transaction.md

This document introduces the principles of TiDB's optimistic transaction model and related features.

> In TiDB's optimistic transaction model, for write-write conflicts, the two-phase commit begins only when the transaction is committed.

Reviewer: The original Chinese is

> 对于写写冲突，只有事务提交时，才会通过两阶段提交检测是否存在写写冲突。

I think the English version is a bit confusing.

Reviewer: ```suggestion
Detecting the write-write conflict will only be triggered when committing the transaction via two-phase commit.
```

> In TiDB's optimistic transaction model, detecting the write-write conflicts only happens when committing the transaction via two-phase commit.

Reviewer 2: ```suggestion
In TiDB's optimistic transaction model, write-write conflicts are detected only at the two-phase transactional commit.
```
